DuckDuckGo/AppServices/AutofillService.swift

```diff
@@ -0,0 +1,104 @@
+
+    var syncService: SyncService?
+
+    init() { // TODO: this was always done after DB was initialized, so looks like this code was never being executed, to confirm
+        let autofillStorage = EmailKeychainManager()
+        if !Database.shared.isDatabaseFileInitialized {
+            try? autofillStorage.deleteAuthenticationState()
+        }
+    }
+
+    func onLaunching() {
+        if AppDependencyProvider.shared.appSettings.autofillIsNewInstallForOnByDefault == nil {
+            AppDependencyProvider.shared.appSettings.setAutofillIsNewInstallForOnByDefault()
+        }
+        autofillPixelReporter = makeAutofillPixelReporter()
+        registerForAutofillEnabledChanges()
+    }
+
+    private func makeAutofillPixelReporter() -> AutofillPixelReporter {
+        AutofillPixelReporter(
+            standardUserDefaults: .standard,
+            appGroupUserDefaults: UserDefaults(suiteName: "\(Global.groupIdPrefix).autofill"),
+            autofillEnabled: AppDependencyProvider.shared.appSettings.autofillCredentialsEnabled,
+            eventMapping: EventMapping<AutofillPixelEvent> { [weak self] event, _, params, _ in
+                switch event {
+                case .autofillActiveUser:
+                    Pixel.fire(pixel: .autofillActiveUser)
+                case .autofillEnabledUser:
+                    Pixel.fire(pixel: .autofillEnabledUser)
+                case .autofillOnboardedUser:
+                    Pixel.fire(pixel: .autofillOnboardedUser)
+                case .autofillToggledOn:
+                    Pixel.fire(pixel: .autofillToggledOn, withAdditionalParameters: params ?? [:])
+                    if let autofillExtensionToggled = self?.autofillUsageMonitor.autofillExtensionEnabled {
+                        Pixel.fire(pixel: autofillExtensionToggled ? .autofillExtensionToggledOn : .autofillExtensionToggledOff,
+                                   withAdditionalParameters: params ?? [:])
+                    }
+                case .autofillToggledOff:
+                    Pixel.fire(pixel: .autofillToggledOff, withAdditionalParameters: params ?? [:])
+                    if let autofillExtensionToggled = self?.autofillUsageMonitor.autofillExtensionEnabled {
+                        Pixel.fire(pixel: autofillExtensionToggled ? .autofillExtensionToggledOn : .autofillExtensionToggledOff,
+                                   withAdditionalParameters: params ?? [:])
+                    }
+                case .autofillLoginsStacked:
+                    Pixel.fire(pixel: .autofillLoginsStacked, withAdditionalParameters: params ?? [:])
+                default:
+                    break
+                }
+            },
+            installDate: StatisticsUserDefaults().installDate ?? Date()
+        )
+    }
+
+    func onForeground() {
+        guard let syncService else {
+            assertionFailure("SyncService must be injected before calling onForeground.")
```

Comment:
I think we need a pixel for this case.

Reply:
We discussed this with @jaceklyp and I agree it's not needed. The assertion here is enough.